packages/bot-skeleton/src/scratch/dbot.js

```diff
@@ -121,13 +122,14 @@ class DBot {
      */
     runBot() {
         try {
+            const api = BinarySocketBase;
+            api.openNewConnection();
             const code = this.generateCode();
-
             if (this.interpreter !== null) {
```

Comment:
```suggestion
            if (this.interpreter) {
```

Reply:
this condition applies when a user stops the bot and then run it again and should check if the interpreter is null or not